src/regex/lib_pcre2.cr

```diff
@@ -194,23 +194,18 @@ lib LibPCRE2
   type JITStack = Void*
 
   fun jit_stack_create = pcre2_jit_stack_create_8(startsize : LibC::SizeT, maxsize : LibC::SizeT, gcontext : GeneralContext) : JITStack
-  fun jit_stack_assign = pcre2_jit_stack_assign_8(mcontext : MatchContext, callable_function : Void*, callable_data : Void*) : Void
+  fun jit_stack_assign = pcre2_jit_stack_assign_8(mcontext : MatchContext, callable_function : Void* -> JITStack, callable_data : Void*) : Void
 
   fun pattern_info = pcre2_pattern_info_8(code : Code*, what : UInt32, where : Void*) : Int
 
-  fun match = pcre2_match_8(code : Code*, subject : UInt8*, length : LibC::SizeT, startoffset : LibC::SizeT, options : UInt32, match_data : MatchData*, mcontext : MatchContext) : Int
-  fun match_data_create_from_pattern = pcre2_match_data_create_from_pattern_8(code : Code*, gcontext : GeneralContext) : MatchData*
-  fun match_data_free = pcre2_match_data_free_8(match_data : MatchData*) : Void
+  fun match = pcre2_match_8(code : Code*, subject : UInt8*, length : LibC::SizeT, startoffset : LibC::SizeT, options : UInt32, match_data : MatchData, mcontext : MatchContext) : Int
+  fun match_data_create_from_pattern = pcre2_match_data_create_from_pattern_8(code : Code*, gcontext : GeneralContext) : MatchData
```

Comment:
`Code` doesn't need a `*` here, as it is already a `Void*` and this isn't an out parameter. (Or better yet, the `type` declarations should use a plain `Void` and the `*`s should be kept in all those funs, to stay consistent with the actual C header file.)